integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ItMiiSampleHelper.java

```diff
@@ -0,0 +1,332 @@
+        outStr += ", command=\n{\n" + command + "\n}\n";
+        outStr += ", stderr=\n{\n" + (result != null ? result.stderr() : "") + "\n}\n";
+        outStr += ", stdout=\n{\n" + (result != null ? result.stdout() : "") + "\n}\n";
+
+        assertTrue(success, outStr);
+      }
+
+      previousTestSuccessful = true;
+    }
+  }
+
+  /**
+   * Test MII sample WLS or JRF initial use case.
+   */
+  public static void callInitialUseCase() {
+    String imageName = (domainType.equals(DomainType.WLS))
+        ? MII_SAMPLE_WLS_IMAGE_NAME_V1 : MII_SAMPLE_JRF_IMAGE_NAME_V1;
+    previousTestSuccessful = true;
+    envMap.put("MODEL_IMAGE_NAME", imageName);
+
+    if (domainType.equals(DomainType.JRF)) {
+      String dbImageName = (KIND_REPO != null
+          ? KIND_REPO + DB_IMAGE_NAME.substring(BASE_IMAGES_REPO.length() + 1) : DB_IMAGE_NAME);
+      String jrfBaseImageName = (KIND_REPO != null
+          ? KIND_REPO + FMWINFRA_IMAGE_NAME.substring(BASE_IMAGES_REPO.length() + 1) : FMWINFRA_IMAGE_NAME);
+      String dbNamespace = envMap.get("dbNamespace");
+
+      envMap.put("DB_IMAGE_NAME", dbImageName);
+      envMap.put("DB_IMAGE_TAG", DB_IMAGE_TAG);
+      envMap.put("DB_NODE_PORT", "none");
+      envMap.put("BASE_IMAGE_NAME", jrfBaseImageName);
+      envMap.put("BASE_IMAGE_TAG", FMWINFRA_IMAGE_TAG);
+      envMap.put("POD_WAIT_TIMEOUT_SECS", "1000"); // JRF pod waits on slow machines, can take at least 650 seconds
+      envMap.put("DB_NAMESPACE", dbNamespace);
+      envMap.put("DB_IMAGE_PULL_SECRET", BASE_IMAGES_REPO_SECRET); //ocr/ocir secret
+      envMap.put("INTROSPECTOR_DEADLINE_SECONDS", "600"); // introspector needs more time for JRF
+
+      // run JRF use cases irrespective of WLS use cases fail/pass
+      previousTestSuccessful = true;
+      execTestScriptAndAssertSuccess(domainType, "-db,-rcu", "DB/RCU creation failed");
+    }
+
+    execTestScriptAndAssertSuccess(
+        domainType,
+        "-initial-image,-check-image-and-push,-initial-main",
+        "Initial use case failed"
+    );
+  }
+
+  /**
+   * Test MII sample WLS or JRF update1 use case.
+   */
+  public static void callUpdateUseCase(String args,
+                                       String errString) {
+    if (args.contains("update3")) {
+      String imageName = (domainType.equals(DomainType.WLS))
+          ? MII_SAMPLE_WLS_IMAGE_NAME_V2 : MII_SAMPLE_JRF_IMAGE_NAME_V2;
+      envMap.put("MODEL_IMAGE_NAME", imageName);
+    }
+
+    execTestScriptAndAssertSuccess(domainType, args, errString);
+  }
+
+  /**
+   * Test MII sample WLS or JRF update1 use case.
+   */
+  public static void callCheckMiiSampleSource(String args,
+                                              String errString) {
+    final String baseImageNameKey = "BASE_IMAGE_NAME";
+    envMap.remove(baseImageNameKey);
+    execTestScriptAndAssertSuccess(domainType, args, errString);
+    envMap.put(baseImageNameKey, WEBLOGIC_IMAGE_NAME);
```

Comment:
minor: This looks like a refactoring issue. If, sometime in the future, the suite was changed so that this test was called from a different location and this is not fixed, then the test would potentially break the next test. The code shouldn't force WEBLOGIC_IMAGE_NAME -- instead, the code should restore the image that was originally in the envMap, similar to the original code, and it should ideally do this in a finally.